beps/docs/proposals/BEP-008-function-and-lambda-syntax-design/README.md

```
@@ -0,0 +1,846 @@
+But in our language, **type names and variable names are always distinguishable**. Built-in types like `int`, `string`, and `bool` are reserved keywords — you cannot name a variable `int`. User-defined types follow a capitalized naming convention (e.g. `Response`, `UserProfile`).
+
+This means the parser can always determine what follows `->`:
+
+```
+(x: int) -> int           // `int` is a type → this is a function type
+(x: int) -> count         // `count` is a variable → this is a lambda
+(x: int) -> int { x + 1 }         // expression with operator → this is a lambda
+(x: int) -> T { x + 1 }    // brace → this is a lambda with block body
+```
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)